server/channels/app/shared_channel.go

```diff
@@ -244,7 +245,16 @@ func (a *App) OnSharedChannelsSyncMsg(msg *model.SyncMsg, rc *model.RemoteCluste
 func (a *App) OnSharedChannelsPing(rc *model.RemoteCluster) bool {
 	pluginHooks, err := getPluginHooks(a.GetPluginsEnvironment(), rc.PluginID)
 	if err != nil {
-		a.Log().Error("Ping for shared channels cannot get plugin hooks", mlog.String("plugin_id", rc.PluginID), mlog.Err(err))
+		// plugin was likely uninstalled. Issue a warning once per hour, with instructions how to clean up if this is
+		// intentional.
+		if time.Now().Minute() == 0 {
+			msg := "Cannot find plugin for shared channels ping; if the plugin was intentionally uninstalled, "
+			msg = msg + "stop this warning using  `/secure-connection remove --connectionID %s`"
```

Comment:
Huh, so I guess an admin could completely break an otherwise stable MS Teams by executing this? Long term, I'm thinking we should aim to decouple "Shared Channels" from "shared channels as a replacement for hooks", otherwise this could be a footgun waiting to happen.

Reply:
Potentially, but restarting the plugin would fix it again as it registers on every OnActivate.